src/main.rs

```diff
@@ -172,23 +181,50 @@ impl App {
             println!("{}", &contents);
             return Ok(())
         }
-        let mut styles = if args.get_flag("no-color") {
+        let mut style = if args.get_flag("no-color") {
             OutputStyle::plain()
         } else {
             OutputStyle::default()
         };
         if let Some(show_ws) = args.get_one::<bool>("show-whitespace") {
             if *show_ws {
-                styles.input_whitespace = styles.input_whitespace.or(Some(styles.input));
-                styles.output_whitespace = styles.output_whitespace.or(Some(styles.output));
+                style.input_whitespace = style.input_whitespace.or(Some(style.input));
+                style.output_whitespace = style.output_whitespace.or(Some(style.output));
             } else {
-                styles.input_whitespace = None;
-                styles.output_whitespace = None;
+                style.input_whitespace = None;
+                style.output_whitespace = None;
             }
         }
         let clash: Clash = serde_json::from_str(&contents)?;
 
-        clash.pretty_print(styles)
+        // Testcases subcommand
+        if let Some(values) = args.get_many::<usize>("testcases") {
+            let testcases_to_print: Vec<usize> = values.cloned().collect();
+
+            // Random default value for "print all"
+            if testcases_to_print[0] == 999 {
+                let _ = clash.print_testcases(style);
+                return Ok(())
+            }
+
+            // println!("{} {:?}", "some args were given", &testcases_to_print);
+            let max_idx = clash.testcases().len() / 2;
+            for testcase_idx in testcases_to_print {
+                if testcase_idx > max_idx {
+                    return Err(anyhow!("Invalid index: {}. The clash only has {} tests.", testcase_idx, max_idx));
+                }
+                // To factor validators and 1-indexing
```

Comment:
I would count validators as test cases (although it's not rare to have the same test case as both test and validator which might be an issue). Either way there should be some way to look at validators too.

Relying on insider knowledge of how the test cases / validators are arranged (the CodinGame convention that every second one is a validator) also seems a bit too magical for my taste, I would rather rely on the `is_validator` field if there is ever a reason to separate the two.

Reply:
For the moment I think imitating CG's feature of not having access to the validators is preferable.

Besides, since ideally validators should be similar to the corresponding test, I think printing them with this flag will be clutter.

They should be supported by the testcase command though.